### PR TITLE
Extend repository schema to include sentence embedder

### PR DESCRIPTION
**Pull request by mehmetcanay**
Extended both SQL and Weaviate repository schemas to include sentence embedders.
Made necessary changes to functions utilizing these classes.
Added a new function allowing the user to filter based on the terminology and model.
Added new tests for each repository for the function retrieving all the stored sentence embedders.

### CODE REVIEW

File: datastew/repository/model.py [hunk 1]
```diff
@@ -18,6 +17,15 @@ def __init__(self, name: str, id: str) -> object:
         self.id = id
 
 
+class SentenceEmbedder(Base):
```
[the remark this answers is not in the document]
Comment (tiadams) [reply datastew/repository/model.py:20]: The embedding model can just be a single string property in the embedding class.

**Review by tiadams — CHANGES_REQUESTED** [review]

File: datastew/embedding.py [hunk 2]
```diff
@@ -31,29 +35,33 @@ def get_embedding(self, text: str, model="text-embedding-ada-002"):
...
+    
+    def get_model_name(self) -> str:
+        return self.model
 
 
 class MPNetAdapter(EmbeddingModel):
     def __init__(self, model="sentence-transformers/all-mpnet-base-v2"):
         logging.getLogger().setLevel(logging.INFO)
-        self.mpnet_model = SentenceTransformer(model)
+        self.model = SentenceTransformer(model)
```
[the remark this answers is not in the document]
Comment (mehmetcanay) [reply datastew/embedding.py]: Addressed the issue by renaming the model attribute of GPT4Adapter to model_name.

File: datastew/repository/model.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mehmetcanay) [reply datastew/repository/model.py]: Refactored the code accordingly.

File: datastew/repository/weaviate.py [hunk 3]
```diff
@@ -58,6 +59,16 @@ def store_all(self, model_object_instances):
...
             self.store(instance)
 
+    def get_all_sentence_embedders(self) -> List[str]:
+        sentence_embedders = set()
+        try:
+            result = self.client.query.get("Mapping", ["hasSentenceEmbedder"]).do()
+            for item in result['data']['Get']['Mapping']:
+                sentence_embedders.add(item["hasSentenceEmbedder"])
+        except Exception as e:
+            raise RuntimeError(f"Failed to fetch terminologies: {e}")
```
Comment (tiadams) [inline datastew/repository/weaviate.py:69]: Why is it failed to fetch terminologies here?

Comment (mehmetcanay) [reply datastew/repository/weaviate.py]: I typo on my side, sorry. Fixed in the latest push.

**Review by tiadams — APPROVED** [review]

**Pull request merged**